### PR TITLE
SOAR_2629_Update_Lookup_Vulnerability_From_Slack

### PR DESCRIPTION
**Pull request by ekim-r7**
## Description
- Update the workflow to work in the cloud

## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

workflow
workflow2

#### Job

execution1
execution2
job

#### Artifact

artifact1

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 24.093ms
```

### CODE REVIEW

File: workflows/Lookup_Vuln_From_Slack/workflow.spec.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ hub_tags:
   keywords: [remediation_management, vulnerability_management, chat, chatops]
   features: []
 resources:
-  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Lookup_Vuln_From_Slack
+  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Lookup_Vulnerability_From_Slack
```
Comment (jschipp-r7) [inline workflows/Lookup_Vuln_From_Slack/workflow.spec.yaml:15]: This needs to be reverted
```suggestion
  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Lookup_Vuln_From_Slack
```

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
The workflow is not consistent on whether the user should use `lookup-vuln` or `lookup_vuln` as the chat command. We use the hyphen `-` syntax in workflow commands because they're more similar to how people write English e.g. "ex-president" “self-expression,” “all-knowing". Will need to update all those references too

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]
Need to address Jon's comment before we can merge

File: workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon [hunk 2]
```diff
@@ -428,346 +487,211 @@
...
+            "defaultImageData": "https://us.cdn-assets.connect.insight.rapid7.com/step-type-icons/automated-decision.svg",
             "connectionType": "NONE"
           },
-          "51cf1985-0451-4f54-8856-7140fc08ce9c": {
-            "nodeId": "51cf1985-0451-4f54-8856-7140fc08ce9c",
-            "name": "Post Results to Slack (multiple results loop)",
-            "type": "action_chatops",
+          "8b5f067d-03aa-4d9e-a06b-5ad7dda77b99": {
+            "nodeId": "8b5f067d-03aa-4d9e-a06b-5ad7dda77b99",
+            "name": "lookup-vulnerability Trigger",
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon:536]:
```suggestion
            "name": "lookup-vuln Trigger",
```

File: workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon [hunk 3]
```diff
@@ -428,346 +487,211 @@
...
             "continueOnFailure": false,
             "isDisabled": false,
             "parameters": {
               "input": {
-                "attachments": [],
-                "message": "Severity: `{{[919c2297-d695-4af5-b29c-d099a0fb07a1].[content_result].[severity]}}`\nhttps://rapid7.com/db/vulnerabilities/{{[58eae3a8-f9f1-4006-b628-05e9cdca896a].[basename]}}",
-                "recipient": "{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[message].[channel_id]}}",
-                "thread_ts": "{{[02bdfb48-2c18-44fb-9b08-527bad014afe].[timestamp]}}"
+                "matchChannel": "change_me",
+                "matchText": "lookup-vulnerability",
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon:544]:
```suggestion
                "matchText": "lookup-vuln",
```

File: workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon [hunk 4]
```diff
@@ -1599,21 +1450,39 @@
...
+        "humanCostDisplayUnit": "minutes"
       }
     ],
     "triggers": [
       {
-        "id": "cd82f809-3b85-4c22-94c5-61e0ac44e5b1",
-        "name": "lookup-vuln Trigger",
-        "description": "Trigger this workflow by entering a Slack command in the following format:\n\"@Rapid7 InsightConnect lookup-vuln CVE-XXXX-XXXX\"\nOR\n\"@Rapid7 InsightConnect lookup-vuln heartbleed\"",
+        "id": "1f531da7-291f-4303-a4fa-a81b804654d3",
+        "name": "lookup-vulnerability Trigger",
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon:1481]:
```suggestion
        "name": "lookup-vuln Trigger",
```

File: workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon [hunk 5]
```diff
@@ -1599,21 +1450,39 @@
...
+        "description": "Trigger this workflow by entering a Slack command in the following format:\n\"@Rapid7 InsightConnect lookup-vulnerability CVE-XXXX-XXXX\"\nOR\n\"@Rapid7 InsightConnect lookup-vulnerability heartbleed\"",
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon:1482]:
```suggestion
        "description": "Trigger this workflow by entering a Slack command in the following format:\n\"@Rapid7 InsightConnect lookup-vuln CVE-XXXX-XXXX\"\nOR\n\"@Rapid7 InsightConnect lookup-vuln heartbleed\"",
```

File: workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon [hunk 6]
```diff
@@ -1599,21 +1450,39 @@
...
         "input": {
-          "matchChannel": "",
-          "matchText": "(?i)lookup-vuln",
+          "matchChannel": "change_me",
+          "matchText": "lookup-vulnerability",
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon:1485]:
```suggestion
          "matchText": "lookup-vuln",
```

File: workflows/Lookup_Vuln_From_Slack/help.md [hunk 7]
```diff
@@ -25,31 +25,29 @@ After import, activate the workflow in order to trigger it.
 
 *This workflow will trigger in any direct messages to your Chatbot **or** any message in a channel directed @ your Chatbot. Note the Chatbot must be in the channel in order to trigger the workflow this way.*
 
-To run the workflow, send a direct message to your InsightConnect Slack Chatbot or @ your Chatbot in a public channel starting with the command `lookup-vuln`.
+To run the workflow, send a direct message to your InsightConnect Slack Chatbot or @ your Chatbot in a public channel starting with the command `lookup-vulnerability`.
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/help.md:28]:
```suggestion
To run the workflow, send a direct message to your InsightConnect Slack Chatbot or @ your Chatbot in a public channel starting with the command `lookup-vuln`.
```

File: workflows/Lookup_Vuln_From_Slack/help.md [hunk 8]
```diff
@@ -25,31 +25,29 @@ After import, activate the workflow in order to trigger it.
...
 
 For example, in a direct message to your Chatbot:
-* `lookup-vuln CVE-2020-0674`
+* `lookup-vulnerability CVE-2020-0674`
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/help.md:31]:
```suggestion
* `lookup-vuln CVE-2020-0674`
```

File: workflows/Lookup_Vuln_From_Slack/help.md [hunk 9]
```diff
@@ -25,31 +25,29 @@ After import, activate the workflow in order to trigger it.
...
 
 Or in a channel including your Chatbot:
-* `@Rapid7 InsightConnect lookup-vuln bluekeep`
+* `@Rapid7 InsightConnect lookup-vulnerability bluekeep`
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/help.md:34]:
```suggestion
* `@Rapid7 InsightConnect lookup-vuln bluekeep`
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]
Left comments for the spots that need to be updated - I'd recommend doing the updates in the workflow builder to play it safe. The following screenshots also need to be updated aftewards: execution1.png, execution2.png, workflow.png

File: workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon [hunk 10]
```diff
@@ -272,11 +272,11 @@
             "isDisabled": false,
             "parameters": {
               "captureAll": false,
-              "expressionText": "lookup-vulnerability\\s+{{argument:/.*/}}",
+              "expressionText": "edit-me\\s+{{argument:/.*/}}",
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon:275]:
```suggestion
              "expressionText": "lookup-vuln\\s+{{argument:/.*/}}",
```

File: workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon [hunk 11]
```diff
@@ -272,11 +272,11 @@
...
               "expressions": [
                 {
                   "type": "text",
-                  "value": "lookup-vulnerability\\s+"
+                  "value": "edit-me\\s+"
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vuln_From_Slack/Lookup_Vuln_From_Slack.icon:279]:
```suggestion
                  "value": "lookup-vuln\\s+"
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Pull request merged**